src/Reduced.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ElegantBro\Arrayee;
+
+use ElegantBro\Interfaces\Arrayee;
+use Exception;
+use function array_merge;
+
+final class Reduced implements Arrayee
+{
+    /**
+     * @var Arrayee
+     */
+    private $arrayee;
+    /**
+     * @var callback
+     */
+    private $callback;
+    /**
+     * @var ?Arrayee
+     */
+    private $initial;
+
+    /**
+     * Reduced constructor.
+     * @param Arrayee $arrayee
+     * @param callable $callback
+     * @param Arrayee|null $initial
+     */
+    public function __construct(Arrayee $arrayee, callable $callback, ?Arrayee $initial = null)
+    {
+        $this->arrayee = $arrayee;
+        $this->callback = $callback;
+        $this->initial = $initial ?? new Just([]);
```

Comment:
I have two points about this part: you shouldn't take this assert in the constructor and shouldn't use nullable arguments

try to create a secondary constructor

Reply:
Yeah, the primary constructor has no nullable arguments. And the secondary constructor will look like `initialEmpty(Arrayee $arrayee, callable $callback)`